.github/workflows/doc.yml

```diff
@@ -18,24 +23,57 @@ jobs:
         uses: mozilla-actions/sccache-action@v0.0.3
         with:
           version: "v0.5.2"
-      - name: Docs
+      - name: build rustdocs
         run: |
-          RUSTDOCFLAGS="--markdown-css rust.css --markdown-no-toc --index-page docs/rustdoc/index.md -Zunstable-options" cargo doc --workspace --no-deps --document-private-items
+          cargo doc --workspace --no-deps --document-private-items
           cp docs/rustdoc/rust.css target/doc/rust.css
-        env:
-          SCCACHE_GHA_ENABLED: "true"
-          RUSTC_WRAPPER: "sccache"
       - name: Fix file permissions
         shell: sh
         run: |
           chmod -c -R +rX "target/doc" |
           while read line; do
               echo "::warning title=Invalid file permissions automatically fixed::$line"
           done
-      - name: Upload artifacts
-        uses: actions/upload-pages-artifact@v1
+      - name: Archive artifact
+        shell: sh
+        if: runner.os == 'Linux'
+        run: |
+          chmod -c -R +rX "$INPUT_PATH" | while read line; do
+            echo "::warning title=Invalid file permissions automatically fixed::$line"
+          done
+          tar \
+            --dereference --hard-dereference \
+            --directory "$INPUT_PATH" \
+            -cvf "$RUNNER_TEMP/artifact.tar" \
+            --exclude=.git \
+            --exclude=.github \
+            .
+        env:
+          INPUT_PATH: target/doc
+      - name: Install cargo-docset
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-docset
+      - name: build docsets
+        run: |
+          cargo docset --no-clean
+          tar -cvzf risingwave.docset.tgz target/docset/risingwave.docset
+
+          echo "<entry>
+          <version>$(date -u +%Y-%m-%dT%H:%M:%SZ)</version>
```

Comment:
https://kapeli.com/docsets#dashdocsetfeed

> You can use any versioning system you want. Dash will use string comparison to determine whether or not to download an update.

So I think timestamp might be good